enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisterAllocation.scala

```diff
@@ -218,13 +218,16 @@ object RegisterAllocation {
         incomingPipeline.newLong(name, nullable = false, CTNode)
         incomingPipeline
 
+      case MergeCreateNode(_, IdName(name), _, _) =>
+        // The variable name should already have been allocated by the NodeLeafPlan
+        incomingPipeline
+
       case CreateRelationship(_, IdName(name), startNode, typ, endNode, props) =>
         incomingPipeline.newLong(name, nullable = false, CTRelationship)
         incomingPipeline
 
       case MergeCreateRelationship(_, IdName(name), startNode, typ, endNode, props) =>
         incomingPipeline.newLong(name, nullable = false, CTRelationship)
-        incomingPipeline
```

Comment:
does `newLong()` return a `PipelineInformation`? This change breaks source code symmetry for me, I think I would prefer to keep this `incomingPipeline` for clarity, or if we remove it, remove it everywhere.